syntax = "proto3";

package events;

import "google/protobuf/timestamp.proto";


enum Result {
    SUCCESS = 0;
    FAILURE = 1;
    ABORTED = 2;
}

message PipelineStarted {
  string uuid = 1;
  google.protobuf.Timestamp timestamp = 2;
}

message PipelineFinished {
  string uuid = 1;
  google.protobuf.Timestamp timestamp = 2;
  int32 duration_ms = 3;
  Result result = 4;
}

message PipelineStageStarted {
  string pipeline_uuid = 1;
  string uuid = 2;
  google.protobuf.Timestamp timestamp = 3;
}

message PipelineStageFinished {
  string pipeline_uuid = 1;
  string uuid = 2;
  google.protobuf.Timestamp timestamp = 3;
  int32 duration_ms = 4;
  Result result = 5;
}

message ScmUser {
    string id = 1;
    string username = 2;
    string email = 3;
}

message ScmCommit {
    string sha = 1;
    string message = 2;
    ScmUser author = 3;
    google.protobuf.Timestamp timestamp = 4;
}

message CodePushed {
    string repository_name = 1;
    string new_head_sha = 2;
    string previous_head_sha = 3;
    ScmUser pusher = 4;
    repeated ScmCommit commits = 5;
    google.protobuf.Timestamp timestamp = 6;
}

// DeployStarted
// DeployFinished
// ArtifactPublished
// PullRequestOpened
// PullRequestClosed
// CodeReviewAccepted
// RepositoryAdded
// RepositoryRemoved
